py3status/modules/xrandr_toggle.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+"""
+Toggle monitor output from the status bar using xrandr.
+
+Configuration parameters:
+    cache_timeout: how often we refresh usage in seconds (default: 10s)
+    distext: custom text or symbol for disconnected status
```

Comment:
One example would be this part:

```
            # if source is set, use that instead of number
            if len(self.source) > 0:
                self.mon = next((x for x in monitors                    
                                if x['id'] == self.source),
                                None)
                if self.mon is None:
                    return self._error_response(i3s_config['color_bad'],
                                                'monitor not found')    
            else:
                self.mon = monitors[number]
```

It looks really confusing to me, I also had `self.monitor` but eventually renamed it to `self.mon` and had to split the loop to fit that limit. The easiest way to avoid having that limitation would have been something like this I guess:

```
            # if source is set, use that instead of number
            if len(self.source) > 0:
                self.monitor = None
                for x in monitors:
                    if x['id'] == self.source:
                        self.monitor = x
                        break
                if self.monitor is None:
                    color_bad = i3s_config['color_bad']
                    message = 'monitor not found'
                    return self._error_response(color_bad, message)
            else:
                self.monitor = monitors[number]
```

Or would you have done this in another manner?